projects/igniteui-angular/src/lib/badge/badge.component.ts

```diff
@@ -69,29 +79,54 @@ export class IgxBadgeComponent {
      * Has priority over the `value` property.
      * If neither a `value` nor an `icon` is set the content of the badge will be empty.
      * Providing an invalid value won't display anything.
+     * ```html
+     *   <igx-badge icon="check" type="success" class="badge-style" value="11"></igx-badge>
+     * ```
      */
     @Input()
     public icon: string;
 
     /**
-     * Set the value of the `role` attribute.
+     * This allows you to set value to role attribute.
+     *```html
+     *  @ViewChild("MyBadge", { read: IgxBadgeComponent })
+     *  public badge: IgxBadgeComponent;
+     *
+     *  //...
+     *  badge.label = "badge-status";
```

Comment:
@alexalex972 The code snippet shows how to set "label" and not "role". The property is "role".